x-pack/plugins/fleet/server/services/epm/elasticsearch/ilm/install.ts

```diff
@@ -5,38 +5,54 @@
  * 2.0.
  */
 
-import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import type { ElasticsearchClient, Logger, SavedObjectsClientContract } from '@kbn/core/server';
 
-import type { InstallablePackage } from '../../../../types';
+import type { EsAssetReference, InstallablePackage } from '../../../../types';
 
 import { ElasticsearchAssetType } from '../../../../types';
 import { getAsset, getPathParts } from '../../archive';
+import { updateEsAssetReferences } from '../../packages/install';
 import { getESAssetMetadata } from '../meta';
 import { retryTransientEsErrors } from '../retry';
 
 export async function installILMPolicy(
   packageInfo: InstallablePackage,
   paths: string[],
   esClient: ElasticsearchClient,
-  logger: Logger
-) {
+  savedObjectsClient: SavedObjectsClientContract,
+  logger: Logger,
+  esReferences: EsAssetReference[]
+): Promise<EsAssetReference[]> {
   const ilmPaths = paths.filter((path) => isILMPolicy(path));
-  if (!ilmPaths.length) return;
-  await Promise.all(
-    ilmPaths.map(async (path) => {
-      const body = JSON.parse(getAsset(path).toString('utf-8'));
+  if (!ilmPaths.length) return esReferences;
+
+  const ilmPolicies = ilmPaths.map((path) => {
+    const body = JSON.parse(getAsset(path).toString('utf-8'));
+
+    body.policy._meta = getESAssetMetadata({ packageName: packageInfo.name });
+
+    const { file } = getPathParts(path);
+    const name = file.substr(0, file.lastIndexOf('.'));
 
-      body.policy._meta = getESAssetMetadata({ packageName: packageInfo.name });
+    return { name, body };
+  });
 
-      const { file } = getPathParts(path);
-      const name = file.substr(0, file.lastIndexOf('.'));
+  esReferences = await updateEsAssetReferences(savedObjectsClient, packageInfo.name, esReferences, {
+    assetsToAdd: ilmPolicies.map((policy) => ({
+      type: ElasticsearchAssetType.ilmPolicy,
+      id: policy.name,
+    })),
+  });
```

Comment:
We weren't previously keeping track of these references so these weren't getting deleted on uninstall. Fixed in this PR.